src/toast/Toast.tsx

```diff
@@ -0,0 +1,58 @@
+import "./_toast.scss";
+
+import React, {useEffect} from "react";
+import ReactDOM from "react-dom";
+import classNames from "classnames";
+
+import {useToast} from "./util/toastHooks";
+
+export interface ToastProps {
+  toastRootId?: string;
+}
+
+function Toast({toastRootId}: ToastProps) {
+  const {
+    toastState: {
+      isDisplayed,
+      data: {autoClose, timeout, content, mode, customClassName}
+    },
+    dispatchToastAction
+  } = useToast();
+  let toastRootNode = document.querySelector(toastRootId || "#toast-root");
+
+  if (!toastRootNode) {
+    toastRootNode = document.createElement("div");
+    toastRootNode.setAttribute("id", "toast-root");
+    document.body.appendChild(toastRootNode);
+  }
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (autoClose) {
+      timeoutId = setTimeout(() => {
+        dispatchToastAction({
+          type: "HIDE"
+        });
+
+        if (toastRootNode) {
+          document.body.removeChild(toastRootNode);
+        }
```

Comment:
Don't we remove the `toastRootNode` the user provided (not generated with line 26) with this code? 🤔  I guess we wouldn't want that

Reply:
Which is `toastRootNode` that the user provided? In this case, we just create a node and then remove this after display the message.

Reply:
Bilge is right. I think we should keep the node if it was provided by the user of the component. BTW, how do I provide `toastRootId` to this component when integrating it to a project ? Also, DOM manipulations within this component should be done within a layoutEffect, and the removal of the node can be handled within that effect's cleanup.